### PR TITLE
[folly, cachelib] support libdwarf related functionality

### PR DESCRIPTION
**Pull request by c8ef**
- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md).
- [x] SHA512s are updated for each updated download.
- [x] The "supports" clause reflects platforms that may be fixed by this new version.
- [x] Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.
- [x] Any patches that are no longer applied are deleted from the port's directory.
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is added to each modified port's versions file.

### CODE REVIEW

**Review by MonicaLiu0311 — APPROVED** [review]

**Comment by MonicaLiu0311** [comment]
folly
-
Feature `libdwarf` is tested successfully in the following triplet:
```
x64-windows
x64-windows-static
```

File: ports/folly/vcpkg.json [hunk 1]
```diff
@@ -48,6 +49,12 @@
       "description": "Support compile with libaio",
       "supports": "linux & x64"
     },
+    "libdwarf": {
+      "description": "Support compile with libdwarf",
+      "dependencies": [
+        "libdwarf"
+      ]
+    },
```
Comment (BillyONeal) [inline ports/folly/vcpkg.json:57]: What hooks up this feature? There should be corresponding changes in portfile.cmake to hook it up but I don't see them...

Comment (dg0yt) [reply ports/folly/vcpkg.json:57]: "Header-based" means auto-detection means installation order problems with binary caching.  
Folly may pick dependencies (libdwarf) which are installed for other reasons then the feature being activated.
Then the cached artifact will not record not that it needs to have the dependency installed first.
The the user will see errors when he tries to use folly.

**Review by BillyONeal — CHANGES_REQUESTED** [review]
Locking whether the feature is enabled to the feature setting needs to be implemented.

Implementing this with a REQUIRE_FIND_PACKAGE/DISABLE_FIND_PACKAGE pair in `vcpkg_check_features` is acceptable.

**Pull request closed without merging**